singularity-eos/eos/modifiers/tilted_maxwell_eos.hpp

```diff
@@ -0,0 +1,314 @@
+    };
+    Real rho_dpdrho_min = 0.0; 
+    RootCounts counts;
+    auto stat = regula_falsi(d2pdrho2, 
+                             0.0, 1.0, 
+                             0.1, 3.0,
+                             tol, tol,
+                             rho_dpdrho_min, counts);
+    if (dpdrho(rho_dpdrho_min) >= 0.0) return std::make_tuple(0.0, 0.0, 0.0, false); 
+
+    // Find spinodal region by finding maximum pressure below and minimum pressure above rho_dpdrho_min
+    Real rho_p_max = 0.0; 
+    Real rho_p_min = 0.0;
+    stat = regula_falsi(dpdrho, 
+                        0.0, 0.8 * rho_dpdrho_min, 
+                        eos_rho_min, rho_dpdrho_min, 
+                        tol, tol,
+                        rho_p_max, counts);
+    stat = regula_falsi(dpdrho, 
+                        0.0, 1.2 * rho_dpdrho_min, 
+                        rho_dpdrho_min, eos_rho_max, 
+                        tol, tol,
+                        rho_p_min, counts);
+    Real p_max = std::min(t_.PressureFromDensityTemperature(rho_p_max, temp), 
+                          t_.PressureFromDensityTemperature(eos_rho_max, temp));
+    Real p_min = std::max(t_.PressureFromDensityTemperature(rho_p_min, temp),
+                          t_.PressureFromDensityTemperature(eos_rho_min, temp));
+
+    // Now need to search in between p_min and p_max for the pressure obeying the Maxwell construction conditions 
+    // This requires a function that takes a p_in, finds a rho_low with p_in = p(rho_low, temp) and a rho_hi 
+    // with p_in = p(rho_hi, temp), then returns the Maxwell construction integral between those two densities.
+    Real p_maxwell;
+    stat = regula_falsi([this, temp, rho_p_max, rho_p_min](double p) -> double {return std::get<0>(this->TestMaxwellConditions(p, temp, rho_p_min, rho_p_max));},
+                        0.0, 0.5 * (p_max + p_min), 
+                        p_min, p_max,
+                        tol, tol, 
+                        p_maxwell, counts);
+    const auto [f, rho_maxwell_low, rho_maxwell_hi] = TestMaxwellConditions(p_maxwell, temp, rho_p_min, rho_p_max); 
+
+    return std::make_tuple(rho_maxwell_low, rho_maxwell_hi, p_maxwell, true);
+  }
+
+  auto GetTiltedMaxwellConstruction(Real temp) const {
+    const auto [rho_maxwell_low, rho_maxwell_hi, p_maxwell, has_spinodal] = this->GetMaxwellConstruction(temp);
+    if (!has_spinodal) return std::make_tuple(0.0, 0.0, 0.0, false); 
+    
+    auto func = [this, temp, p_maxwell](std::array<Real, 2> x){
+      const Real Pl = this->t_.PressureFromDensityTemperature(x[0], temp); 
+      const Real Ph = this->t_.PressureFromDensityTemperature(x[1], temp);
+      const Real PD = math_utils::GLQuad32([this, temp](Real rho){return this->t_.PressureFromDensityTemperature(rho, temp)/(rho * rho);}, x[0], x[1])/(1/x[0] - 1/x[1]);
+      return std::array<Real, 2>{(Pl - PD * this->gp(1.0)) / p_maxwell, (Ph - PD * this->gp(0.0)) / p_maxwell};
+    };
+    
+    // NR Iteration
```

Comment:
Should we have a 2D NR somewhere global?

Reply:
Maybe, I have not robustified this yet and it may be that there are some problem specific things that can be done.